src/Emulator/EmulatorProvider.tsx

```diff
@@ -24,8 +24,8 @@ import { DeviceInterface } from 'src/common/types';
 import { useTopology, TopologyState, TopologyActions, Connection } from 'src/Emulator/useTopology';
 
 interface DialogueInterface {
-  dialogueMessages: string[];
-  appendDialogue: (message: string) => any;
+  dialogueMessages: string[][];
```

Comment:
I think it would be more extensible to create a new interface like below instead of an array.

```ts
interface DialogueMessage {
  message: string;
  color: string;
}
...
// and then use
interface DialogueInterface {
  dialogueMessages: DialogueMessage[];
  ...
}
```